### PR TITLE
[Dev] Move to Storybooks 3.2

### PR DESCRIPTION
**Pull request by orta**
Updates the storybook server to handle hierarchy, and removes the need to include your storybooks in the app - the list is pre-generated now.

screen shot 2017-07-31 at 15 44 44

I'll update the code extension after. I handle reaction.

### CODE REVIEW

File: storybook/storyLoader.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+
+// template for doT (https://github.com/olado/doT)
```
Comment (orta) [inline storybook/storyLoader.js]: Fixed with https://github.com/elderfo/react-native-storybook-loader/pull/46

File: package.json [hunk 2]
```diff
@@ -7,79 +7,59 @@
...
     "open-emission": "xcrun simctl launch booted net.artsy.Emission",
+
     "test": "jest",
     "testing": "jest --watch",
     "tslint": "tslint",
+
     "sync-colors": "cd externals/elan && git fetch && git checkout origin/master && cp components/lib/variables/colors.json ../../data",
     "sync-externals": "npm run-script sync-schema && npm run-script sync-colors",
     "sync-schema": "cd externals/metaphysics && git fetch && git checkout origin/master && yarn install && npm run dump-schema -- ../../data && rm -rf node_modules",
+
```
Comment (orta) [inline package.json]: prettier + JSON allows a space. So I've chunked similar commands

File: package.json [hunk 3]
```diff
@@ -7,79 +7,59 @@
...
-      "./src/setupJest.ts"
-    ],
-    "cacheDirectory": ".jest/cache"
-  },
-  "graphql": {
-    "file": "data/schema.json",
-    "tsInterfaceName": "RelayProps"
+    "postinstall": "sed -i '' 's/#import <RCTAnimation\\/RCTValueAnimatedNode.h>/#import \"RCTValueAnimatedNode.h\"/' ./node_modules/react-native/Libraries/NativeAnimation/RCTNativeAnimatedNodesManager.h || true",
+
+    "prestorybook": "rnstl"
```
Comment (orta) [inline package.json]: The thing that generates our list of files for storybooks

File: package.json [hunk 4]
```diff
@@ -130,11 +111,30 @@
   "peerDependencies": {
     "babel-relay-plugin": "^0.9.0"
   },
+  "jest": {
```
Comment (orta) [inline package.json]: moved all configs below the dependencies

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  New dependencies added: react-native-storybook-loader.
  </td>
    </tr>
  </tbody>
</table>



<h2><a href="https://github.com/elderfo/react-native-storybook-loader#readme">react-native-storybook-loader</a></h2>
<p>Author: Chris Gestred</p>
<p>Description: A CLI for dynamically importing stories into [react-native-storybook](https://github.com/storybooks/react-native-storybook).</p>
<p>Homepage: <a href="https://github.com/elderfo/react-native-storybook-loader#readme">https://github.com/elderfo/react-native-storybook-loader#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>5 months ago</td></tr><tr><td>Last Updated</td><td>17 minutes ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>14</td></tr><tr><td>Direct Dependencies</td><td><a href='http: //npmjs.com/package/colors'>colors</a>, <a href='http: //npmjs.com/package/dot'>dot</a>, <a href='http: //npmjs.com/package/findup'>findup</a>, <a href='http: //npmjs.com/package/glob'>glob</a> and <a href='http: //npmjs.com/package/yargs'>yargs</a></td></tr>
</table>

<details>
<summary><code>README</code></summary>
# react-native-storybook-loader

A CLI for dynamically importing stories into [react-native-storybook](https://github.com/storybooks/react-native-storybook).

[Build Status](https://travis-ci.org/elderfo/react-native-storybook-loader) [Known Vulnerabilities](https://snyk.io/test/github/elderfo/react-native-storybook-loader)
[AppVeyor](https://ci.appveyor.com/project/elderfo/react-native-storybook-loader)

## Purpose

While using storybook for React Native, I repeatedly found myself manually creating a file with imports for all my stories. So I built an automated way to do it. `react-native-storybook-loader` can be run using configuration in your `package.json` or via the CLI interface and it generates a file with all the imports for you.

## Installation

```bash
yarn add react-native-storybook-loader -D
```

Or

```bash
npm install react-native-storybook-loader --save-dev
```

## Quick Start

Create a React Native project using [create-react-native-app](https://facebook.github.io/react-native/blog/2017/03/13/introducing-create-react-native-app.html)

```bash
create-react-native-app AwesomeProject
```

Add react-native-storybook to the project using [`getstorybook`](https://getstorybook.io/docs/react-storybook/basics/quick-start-guide)

```bash
cd AwesomeProject
getstorybook
```

Install react-native-storybook-loader

```bash
yarn add react-native-storybook-loader -D
```

Update `index.js` file in the `./storybook` directory to point to `storyLoader.js` (this file will be generated before launching Storybook).

```javascript
import { getStorybookUI, configure } from '@storybook/react-native';
import { loadStories } from './storyLoader';

configure(() => {
  loadStories()
}, module);

const StorybookUI = getStorybookUI({ port: 7007, host: 'localhost' });
export default StorybookUI;
```

Add the `rnstl` cli to the scripts tag of the `package.json`

```json
{
  "scripts": {
    "prestorybook": "rnstl"
  }
}
```

Run Storybook

```bash
yarn storybook
```

Or

```bash
npm run storybook
```

Run react-native on the targeted platform

```bash
react-native run-android
```

Or

```bash
react-native run-ios
```

_Note: If you have problems connecting from your device to Storybook using Android issue the following command: `adb reverse tcp:7007 tcp:7007`._

## Configuration

Story loading is controlled by the `react-native-storybook-loader` section of the project's `package.json`. 

### Options

| Setting | CLI Option | Type | Description | Default | 
|---|---|---|---|---|
| **searchDir** | `--searchDir` | `string` or `string[]` | The directory or directories, relative to the project root, to search for files in. | Project root |
| **outputFile** | `--outputFile` | `string` | The output file that will be written. It is relative to the project directory. | `./storybook/storyLoader.js` | 
| **pattern** | `--pattern` | `string` | The pattern of files to look at. It can be a specific file, or any valid glob. Note: if using the CLI, globs with `**/*...` must be escaped with quotes | `./storybook/stories/index.js` (The default React Native storybook file) | 
|  | `--silent` | | Silences output. This option is not supported in the `package.json` file. | 

> Note: When using the CLI, any of option passed will override the values in the `package.json`

### Examples:

Both examples below will search `src` and `packages` directories recursively for files that end with `.stories.js` and write the output to `./storybook/storyLoader.js`

#### `package.json`

```json
{
  "name": "AwesomeProject",
  "scripts": {
    "prestorybook": "rnstl"
  },
  "config": {
    "react-native-storybook-loader": {
      "searchDir": ["./src", "./packages"],
      "pattern": "**/*.stories.js",
      "outputFile": "./storybook/storyLoader.js"
    }
  }
}
```

#### CLI

**Breaking Change**

CLI can now be accessed from a terminal 
```bash
./node_modules/.bin/rnstl <options>
```
or in package.json 
```json
{
  "scripts": {
    "prestorybook": "rnstl <options>"
  }
}
```

_Note:_ When using a glob with `**/*` it is required to be wrapped in quotes

There is no longer a need to use `node ./node_modules/.bin/rnstl
<options>`.

```bash
$ ./node_modules/.bin/rnstl --searchDir ./src ./packages --pattern "**/*.stories.js" --outputFile ./storybook/storyLoader.js
```

## Support

Please log issues

## Contributing

Coming Soon

</details>




  <details>
    <summary><code>yarn why react-native-storybook-loader</code> output</summary>
    <p><code><ul><li>Has been hoisted to "react-native-storybook-loader"</li><li>This module exists because it's specified in "devDependencies".</li><li>Disk size without dependencies: "980kB"</li><li>Disk size with unique dependencies: "1.84MB"</li><li>Disk size with transitive dependencies: "2.5MB"</li><li>Amount of shared dependencies: 24
    </li></ul></code></p>
  </details>
  
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: package.json (hunk 3 above)
Comment (alloy) [inline package.json]: Like a cat walking over a keyboard.

File: package.json (hunk 4 above)
Comment (alloy) [inline package.json]: YES!

File: storybook/storyLoader.js (hunk 1 above)
Comment (alloy) [inline storybook/storyLoader.js]: ~~I don’t really understand this part.~~

Ohhhh, I see. This is autogenerated 👍

**Comment by alloy** [comment]
Dope 👍

**Pull request merged**